move/committee/sources/committee.move

```diff
@@ -0,0 +1,640 @@
+
+            // Transfer the key server object to the Committee object via TTO
+            transfer::public_transfer(key_server, committee.id.to_address());
+            committee.state = State::Finalized;
+        },
+        _ => abort EInvalidState
+    }
+}
+
+/// Finalize rotation - transfers KeyServer from old committee and destroys the old committee
+public fun finalize_committee_for_rotation(
+    committee: &mut Committee,
+    mut old_committee: Committee,  // Pass by value to destroy it
+    key_server: Receiving<KeyServer>,
+    ctx: &mut TxContext,
+) {
+    assert!(committee.members.contains(&ctx.sender()), ENotMember);
+    assert!(committee.old_committee_id.is_some(), EInvalidState);
+    assert!(object::id(&old_committee) == *committee.old_committee_id.borrow(), EInvalidState);
+
+    // Verify old committee is finalized
+    assert!(match (&old_committee.state) {
+        State::Finalized => true,
+        _ => false,
+    }, ENotFinalized);
+
+    match (&committee.state) {
+        State::PostDKG { approvals, partial_pks, pk: _pk } => {
+            assert!(approvals.length() >= committee.threshold as u64, EInvalidThreshold);
+
+            // Receive the KeyServer from old committee
+            let mut key_server = transfer::public_receive(&mut old_committee.id, key_server);
+
+            // Remove all old partial key servers
+            let mut i = 0;
+            while (i < old_committee.members.length()) {
+                key_server::remove_partial_key_server(&mut key_server, old_committee.members[i]);
+                i = i + 1;
+            };
+
+            // Add partial key servers with new pks
+            let mut j = 0;
+            while (j < committee.members.length()) {
+                key_server::add_partial_key_server(
+                    &mut key_server,
+                    committee.members[j],
+                    partial_pks[j],
+                    ctx,
+                );
+                j = j + 1;
+            };
+
+            // Transfer KeyServer to new committee
+            transfer::public_transfer(key_server, committee.id.to_address());
+
+            // Finalize new committee (master pk is preserved in KeyServer)
+            committee.state = State::Finalized;
+
+            // Destroy old committee
+            let Committee { id, threshold: _, members: _, state: _, old_committee_id: _, old_threshold: _ } = old_committee;
+            object::delete(id);
+        },
+        _ => abort EInvalidState
+    }
+}
+
+/// Update the url of the key server object. Only the 
+public fun update_url(
+    ks: Receiving<KeyServer>,
+    committee: &mut Committee,
+    url: String,
+    ctx: &mut TxContext,
+) {
+    assert!(committee.members.contains(&ctx.sender()), ENotMember);
+    let mut key_server = transfer::public_receive(&mut committee.id, ks);
+    key_server::update_url(&mut key_server, url, ctx);
+    transfer::public_transfer(key_server, committee.id.to_address());
+}
+
```

Comment:
let's add a TODO to handle package upgrades given an approval of threshold of the members